test/testing_utils.py

```diff
@@ -0,0 +1,411 @@
+        if has_lambdaStar and not isinstance(self.lambdaStar, list):
+            self.lambdaStar = [self.lambdaStar]
+
+        if not partial_x:
+            dist = []
+            for x in self.xStar:
+                dist.append(get_dict_distance(x, sol.xStar))
+            self.sol_index = dist.index(min(dist))
+        else:
+            # assume we have a single solution
+            self.sol_index = 0
+        # now we assert against the closest solution
+        # objective
+        # sol.fStar was broken for earlier versions of SNOPT
+        if self.optName == "SNOPT" and Version(self.optVersion) >= Version("7.7.7"):
+            assert_allclose(sol.fStar, self.fStar[self.sol_index], atol=tol, rtol=tol)
+        sol_objectives = np.array([sol.objectives[key].value for key in sol.objectives])
+        assert_allclose(sol_objectives, self.fStar[self.sol_index], atol=tol, rtol=tol)
+        # x
+        assert_dict_allclose(sol.xStar, self.xStar[self.sol_index], atol=tol, rtol=tol, partial=partial_x)
+        dv = sol.getDVs()
+        assert_dict_allclose(dv, self.xStar[self.sol_index], atol=tol, rtol=tol, partial=partial_x)
+        # lambda
+        if (
+            hasattr(self, "lambdaStar")
+            and hasattr(sol, "lambdaStar")
+            and self.lambdaStar is not None
+            and sol.lambdaStar is not None
+        ):
+            assert_dict_allclose(sol.lambdaStar, self.lambdaStar[self.sol_index], atol=tol, rtol=tol)
+
+    def assert_inform_equal(self, sol, optInform=None):
+        """
+        Check that the optInform stored in the Solution object is as expected.
+
+        Parameters
+        ----------
+        sol : Solution object
+            The solution object after optimization
+        optInform : int, optional
+            The expected inform. If None, the default inform is used, which corresponds to
+            a successful optimization termination.
+        """
+        if optInform is not None:
+            self.assertEqual(sol.optInform["value"], optInform)
+        else:
+            # some optimizers do not have informs
+            if self.optName in SUCCESS_INFORM:
+                self.assertEqual(sol.optInform["value"], SUCCESS_INFORM[self.optName])
+
+    def update_OptOptions_output(self, optOptions):
+        """
+        A general helper function to set a default output file name in optOptions
+        based on the optimizer and test method name.
+
+        Parameters
+        ----------
+        optOptions : dict
+            The optimizer options dictionary
+
+        Returns
+        -------
+        optOptions: dict
+            The same optOptions dictionary, but with the output file names set.
+        """
+        optionName = OUTPUT_FILENAMES[self.optName]
+        for optionName, suffix in OUTPUT_FILENAMES[self.optName].items():
+            optOptions[optionName] = f"{self.name}_{self._testMethodName}{suffix}"
```

Comment:
Yeah I know what you mean, these are coming from `unittest`. Will do.

Reply:
I ended up changing these to `self.id()` since that's a public API so hopefully more stable. It's also more explicit even though it will generate some pretty long file names.